File: BineClient/Sdk/app/src/main/java/com/msr/bine_android/utils/BineNotificationHelper.java
package com.msr.bine_android.utils;

import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;

import com.msr.bine_android.R;
import com.msr.bine_sdk.notifications.NotificationHelper;

public class BineNotificationHelper extends NotificationHelper {
    public BineNotificationHelper(@NonNull Context context) {
        super(context);
    }

    @NonNull
    @Override
    public String getMetaResourceBaseUrl() {
        return "https://stcontentcdnstage.blob.core.windows.net/";
    }

    @Override
    public int getAppIcon() {
        return R.drawable.ic_app_icon;
    }

    @NonNull
    @Override
    public Intent getHomeLaunchIntent() {
        return null;
    }
}

File: BineClient/App/BineClient/app/src/main/java/com/microsoft/mobile/polymer/mishtu/utils/BineNotificationHelper.kt
package com.microsoft.mobile.polymer.mishtu.utils

import android.content.Context
import android.content.Intent
import com.microsoft.mobile.polymer.mishtu.BuildConfig
import com.microsoft.mobile.polymer.mishtu.R
import com.microsoft.mobile.polymer.mishtu.ui.activity.MainActivity
import com.msr.bine_sdk.notifications.NotificationHelper

class BineNotificationHelper(val context: Context): NotificationHelper(context){

    override fun getAppIcon(): Int {
        return R.drawable.bn_app_logo
    }

    override fun getMetaResourceBaseUrl(): String {
        return BuildConfig.CDN_BASE_URL
    }

    override fun getHomeLaunchIntent(): Intent {
        val openAppIntent = Intent(context, MainActivity::class.java)
        openAppIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
        openAppIntent.action = MainActivity.NOTIFICATION_TAPPED_BY_USER
        openAppIntent.putExtra(MainActivity.NOTIFICATION_TYPE, BNConstants.NotificationType.Downloads.name)
        return openAppIntent
    }
}